client/signup/main.jsx

```diff
@@ -488,7 +487,13 @@ class Signup extends Component {
 			} );
 		}
 
-		if ( ! userIsLoggedIn && config.isEnabled( 'oauth' ) ) {
+		const isRegularOauth2ClientSignup =
+			dependencies.oauth2_client_id && ! progress?.[ 'oauth2-user' ]?.service; // service is set for social signup (e.g. Google, Apple)
```

Comment:
Great to have these comments here! It was difficult trying to understand all the logic around this. Thanks! ❤️